logmerger/file_reading.py

```diff
@@ -253,3 +256,44 @@ def reader_guard(rdr):
 
     def _close_reader(self):
         self._close_obj.close()
+
+
+class JsonLFileReader(FileReader):
+    @classmethod
+    def _can_read(cls, fname: str) -> bool:
+        return fname.endswith(".jsonl")
+
+    def __init__(self, fname: str, encoding: str):
+        super().__init__(fname, encoding)
+        self._close_obj = open(fname, encoding=encoding, newline='')
+
+        self._iter = self.iter_file()
+
+    @staticmethod
+    def _find_dt_col(d: dict[str, Any], previous_key: str | None):
+        if previous_key is not None:
+            value = d.get(previous_key)
+            if value is not None:
+                return previous_key, value
+        for key, val in d.items():
+            try:
+                tt = TimestampedLineTransformer.make_transformer_from_sample_line(val + " ")
```

Comment:
What is the purpose of adding the space? If there is a value that is not a str in the log line before the timestamp is found, this will raise a TypeError.